FIX #20 / Readability changes, moving alias matching to it's own function

kawatzaki opened the pull request:
Check on #20 for details on the error fixed by this PR.

Improved readability of the code. Please let me know to what extent this is OK.

Commit:
fixes #20

Commit:
added explanation for this fork on readme.md

ibc commented:
Hi @charlesguse, any change to merge this PR and release a new version? Indeed it fixes an important issue.

Commit:
fixed 'default' script handling

Commit:
updated version to 1.0.9

Commit:
fixed version number to match npm package

jimcatts commented on index.js:
```diff
@@ -2,81 +2,113 @@
+   * Execute the regular expression to help identify missing scripts
+   * It also tests for different aliases
+   */
+  osCommand = matchScript(argument, platform, scripts) || matchScript(event, platform, scripts);
```
In the case of lifecycle events, npm_config_argv.original will always be the arguments of the entry point. So for instance when running npm version patch, the version script will be executed with npm_config_argv.original = ['version', 'patch'] and npm_lifecycle_event = 'version' then the postversion script will be executed with npm_config_argv.original = ['version', 'patch'] and npm_lifecycle_event = 'postversion'. In the second case, it will re-run the existing version script during postversion because npm_config_argv.original takes presidence here. I believe this should instead be 
`matchScript(event, platform, scripts) || matchScript(argument, platform, scripts); `
which still should be fine for simple npm run use cases because npm_lifecycle_event is empty and won't match in those cases.

jimcatts committed:
Give npm_lifecycle_event priority over npm_config_argv to support lifecycle events.
Add --no-arguments option to fix https://github.com/charlesguse/run-script-os/issues/23

jimcatts committed:
Fix issue with platform switch falling through and overriding selected script

jimcatts committed:
Fix issue with platform switch falling through and overriding selected script

sators committed:
Add support for npm test shortcut "t"

sators committed:
Add support for npm test shortcut "tst"

sators committed:
Update documentation

sators committed:
Fix index logic for test shortcuts

Commit:
Merge branch 'jimcatts-master'

Commit:
simple style fixes here and there...

Commit:
Merge branch 'feature/npmTest' of https://github.com/sators/run-script-os into sators-feature/npmTest

Commit:
Merge branch 'sators-feature/npmTest'

Commit:
updated version number, increased on feature level

Commit:
fixed let osCommand being duplicated

Commit:
upgraded version to 1.2.1

Commit:
streamlining shorthand expansion

ZChenzy commented:
When will this be merged? The current library is currently not usable for me but the forked repo works perfectly.

charlesguse reviewed (approved):
Thanks for these changes
